rcgen/src/certificate.rs

```diff
@@ -24,30 +24,18 @@ use crate::{
 /// An issued certificate together with the parameters used to generate it.
 #[derive(Debug, Clone)]
 pub struct Certificate {
-	pub(crate) params: CertificateParams,
-	pub(crate) subject_public_key_info: Vec<u8>,
 	pub(crate) der: CertificateDer<'static>,
 }
 
 impl Certificate {
-	/// Returns the certificate parameters
-	pub fn params(&self) -> &CertificateParams {
-		&self.params
-	}
-	/// Calculates a subject key identifier for the certificate subject's public key.
-	/// This key identifier is used in the SubjectKeyIdentifier X.509v3 extension.
-	pub fn key_identifier(&self) -> Vec<u8> {
```

Comment:
I think dropping this fn from the `Certificate` means external users don't have a way to compute a SKI themselves.

That feels reasonable to me, since you could use `x509-parser` for that job, but since it's something the API exposed before I thought it worth calling out the removal.

Reply:
I moved `Certificate::key_identifier()` to `CertificateParams::key_identifier()` (which takes an `&impl PublicKeyData`).

Reply:
SGTM, thanks